src/Npgsql/Internal/TypeHandlers/JsonHandler.cs

```diff
@@ -71,6 +75,20 @@ protected internal override int ValidateAndGetLengthCustom<TAny>([DisallowNull]
                 parameter.ConvertedValue = data;
             return lengthCache.Set(data.Length + _headerLen);
         }
+        
+#if NET6_0_OR_GREATER || NETSTANDARD2_0 || NETSTANDARD2_1
+        if (typeof(TAny) == typeof(JsonObject) || typeof(TAny) == typeof(JsonArray))
```

Comment:
Does PG support a single (scalar) JSON value as a JSON document? If so, should we also support JsonValue (any reason not to)?

Reply:
Nope, it doesn't, it's going to throw: "22P02: invalid input syntax for type json".